src/python/WMCore/FwkJobReport/XMLParser.py

```diff
@@ -334,16 +339,54 @@ def perfSummaryHandler():
     while True:
         report, node = (yield)
         summary = node.attrs.get('Metric', None)
+        module = node.attrs.get('Module', None)
         if summary is None:
             continue
+        site = None
+        if module == 'XrdSiteStatistics':
+            site = summary
+            summary = 'XrdSiteStatistics'
+
         # Add performance section if it doesn't exist
         if not hasattr(report, summary):
             report.section_(summary)
         summRep = getattr(report, summary)
 
         for subnode in node.children:
-            setattr(summRep, subnode.attrs['Name'],
-                    subnode.attrs['Value'])
+#             setattr(summRep, subnode.attrs['Name'],
+#                     subnode.attrs['Value'])
+            name = subnode.attrs['Name']
+            value = subnode.attrs['Value']
+            if isinstance(value, str):
+                # strip off leading and trailing spaces from string values to allow proper data-type casting
+                value = value.lstrip().rstrip()
+            if value == 'false':
+                value = False
+            elif value == 'true':
+                value = True
+            elif pat_float.match(value):
+                value = float(value)
+            elif pat_int.match(value):
+                value = int(value)
+            if module == 'XrdSiteStatistics':
```

Comment:
Unless there is the need to pass many variables around, I would say that this special case deserves its own method/function and a proper documentation.

Reply:
done